### PR TITLE
feat(firebase/rules): setup testing for firebase security rules

### PR DESCRIPTION
**Pull request by d3lm**
Addresses #688

### CODE REVIEW

File: firebase/rules/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "firebase",
```
Comment (PascalPrecht) [inline firebase/rules/package.json]: This is should be renamed to `@machinelabs/firebase-rules` as discussed in https://github.com/machinelabs/machinelabs/pull/699#discussion_r164301875

File: firebase/rules/tests/users.spec.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  it('Current user cannot use a different email', () => {
+    expect(currentUser).cannotWrite('/users/1/common/email', 'new@mail.com');
+  });
+
+  it('Current user cannot change his id', () => {
+    expect(currentUser).cannotWrite('/users/1/common/id', '3');
+  });
+
+  it('Current user cannot update other profiles', () => {
```
Comment (PascalPrecht) [inline firebase/rules/tests/users.spec.ts]: This spec isn't much different from what's already tested in line 58 right?

File: firebase/rules/tests/users.spec.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    expect(currentUser).cannotWrite('/users/2/common/displayName', 'FooBar');
+  });
+
+  it('Users can only read but not update their own plan', () => {
+    const newPlan = { plan_id: 'admin' };
+
+    expect(currentUser).canRead('/users/1/plan');
+    expect(currentUser).cannotWrite('/users/1/plan', newPlan);
+  });
```
Comment (PascalPrecht) [inline firebase/rules/tests/users.spec.ts]: We should add one more test here that a user can also not read the plan of any other users.

File: firebase/rules/tests/users.spec.ts (hunk 2 above)
Comment (d3lm) [reply firebase/rules/tests/users.spec.ts]: True. Was a left-over. Thanks for pointing it out.

File: firebase/rules/package.json (hunk 1 above)
Comment (d3lm) [reply firebase/rules/package.json]: Yep, forgot that.

**Comment by PascalPrecht** [comment]
This LGTM now

**Comment by cburgdorf** [comment]
Great to have this finally working!

**Comment by d3lm** [comment]
Yay 🎉

**Pull request merged**